divi/src/kernel.cpp

```diff
@@ -498,7 +498,7 @@ bool CheckProofOfStake(const CBlock& block, int blockHeight, uint256& hashProofO
     }
 
     //verify signature and script
-    if (!VerifyScript(txin.scriptSig, txPrev.vout[txin.prevout.n].scriptPubKey, STANDARD_SCRIPT_VERIFY_FLAGS, TransactionSignatureChecker(&tx, 0)))
+    if (!VerifyScript(txin.scriptSig, txPrev.vout[txin.prevout.n].scriptPubKey, POS_SCRIPT_VERIFY_FLAGS, TransactionSignatureChecker(&tx, 0)))
```

Comment:
At this point, it would still fail to verify vaults, since the SCRIPT_REQUIRE_COINSTAKE would not be included

Reply:
Yes, but it would fail later (after activation of the fork) when verifying the transaction again inside `main.cpp`.  This is were the fork will be enforced (at least initially, as mentioned in my comment above, we can after the fork simplify some of this).

At this particular place, only coinstake transactions are verified anyway, so it doesn't matter if we enforce the opcode or not.  But if we were to add `SCRIPT_REQUIRE_COINSTAKE` here, the issue would be that it would *accept* `OP_REQUIRE_COINSTAKE` before the fork which the old code does not (since it does not accept NOPs in this particular place).